src/jsil/parser.y

```diff
@@ -132,9 +132,9 @@ procedure_decl: TOK_PROCEDURE proc_ident '(' parameters_opt ')'
               {
                 symbol_exprt proc(to_symbol_expr(parser_stack($2)));
                 code_typet::parameterst parameters;
-                forall_operands(it, parser_stack($4))
+                for(const auto &op : as_const(parser_stack($4)).operands())
```

Comment:
This deviates from the pattern elsewhere -- why the `as_const`?

Reply:
Because `.operands()` would trigger a(n unnecessary) write on the underlying irep, temporarly breaking (probably non-existing) sharing.